### PR TITLE
[DONE] 2D and gw initial waterlevel support

### PR DESCRIPTION
**Pull request by jpprins1**
- [x] Needs update of threedi-api-client

### CODE REVIEW

File: threedi_modelchecker/simulation_templates/models.py [hunk 1]
```diff
@@ -183,15 +181,98 @@ async def save_to_api(self, client: V3BetaApi, simulation: Simulation):
                     simulation_pk=simulation.id, data=self.constant_gw
                 )
             )
-        if self.predefined_1d is not None:
+        initial_waterlevels: List[InitialWaterlevel] = []
+        rasters_lookup: Dict = {}
+
+        # TODO: support user uploaded 1d/2d/gw initial waterlevel file/rasters to threedimodels.
```
Comment (caspervdw) [inline threedi_modelchecker/simulation_templates/models.py:187]: I don't think that is really possible; this module only reads sqlites and everything attached to that.

File: threedi_modelchecker/simulation_templates/models.py [hunk 2]
```diff
@@ -183,15 +181,98 @@ async def save_to_api(self, client: V3BetaApi, simulation: Simulation):
...
+
+        if (
+            self.raster_2d is not None
+            or self.raster_gw is not None
+            or self.waterlevel_1d_file is not None
+        ):
+            # Fetch initial waterlevels
+            initial_waterlevels = (
+                await client.threedimodels_initial_waterlevels_list(
+                    simulation.threedimodel_id
```
Comment (caspervdw) [inline threedi_modelchecker/simulation_templates/models.py:197]: Do we need to order by something here? Or is that the default of the API?

**Review by caspervdw — APPROVED** [review]

File: threedi_modelchecker/simulation_templates/models.py (hunk 1 above)
Comment (jpprins1) [reply threedi_modelchecker/simulation_templates/models.py:187]: You can create SimulationTemplate object programatically or from a dict.
I'm hoping that Lutra is also able to use this part (or maybe even also the sqlite extraction part) to create simulation templates.

File: threedi_modelchecker/simulation_templates/models.py (hunk 2 above)
Comment (jpprins1) [reply threedi_modelchecker/simulation_templates/models.py:197]: For the amount of possible initial waterlevels, this is not needed right now.

**Pull request merged**